src/test_workflow/test_recorder/test_recorder.py

```diff
@@ -95,10 +95,12 @@ class LocalClusterLogs(LogRecorder):
 
     def __init__(self, parent_class: TestRecorder) -> None:
         self.parent_class = parent_class
+        self.number_of_nodes = 0
 
     def save_test_result_data(self, test_result_data: TestResultData) -> None:
         base = self.parent_class._create_base_folder_structure(test_result_data.component_name, test_result_data.component_test_config)
-        dest_directory = os.path.join(base, "local-cluster-logs")
+        dest_directory = os.path.join(base, "local-cluster-logs/id-" + str(self.number_of_nodes))
+        self.number_of_nodes += 1
```

Comment:
The problem with this design as per your log, is that once with-security test complete, the id will not reset and result in id-2/id-3 in without-security dir.

Would suggest reset the number after each run. So it will results in this:
```
with-security/id-0 id-1
without-security/id-0 id-1
```

instead of:
```
with-security/id-0 id-1
without-security/id-2 id-3
```

Reply:
Tests are triggered separately for with and without security so we should be good I believe.

Reply:
The current behavior continues to increase the id throughout different components.